src/target_dir.rs

```diff
@@ -36,38 +40,28 @@ impl CargoDirectories {
     }
 
     pub fn new(cargo_executable: &str) -> CargoDirectories {
-        let manifest_dir = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+        let mut manifest_dir = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
 
-        // Launch 'cargo metadata' pessimistically
-        let mut child = Command::new(cargo_executable)
-            .current_dir(&manifest_dir)
-            .args(["metadata", "--no-deps", "--format-version=1"])
-            .stdout(Stdio::piped())
-            .spawn()
-            .expect("Should be able to run cargo metadata");
-        // Then see if we can find the target directory ourselves
-        let direct_target = manifest_dir.join("target");
-        if direct_target.exists() {
-            let _ = child.kill();
-            return CargoDirectories {
-                target_directory: direct_target,
-                workspace_root: manifest_dir,
-            };
-        }
-        if let Some(parent) = manifest_dir.parent() {
-            let parent_target = parent.join("target");
-            if parent_target.exists() {
-                let _ = child.kill();
+        // First try to find the target directory ourselves
+        // Its possible for this to return false positives if the user leaves a directory named target in the wrong spot.
+        loop {
+            let target = manifest_dir.join("target");
+            if target.exists() {
                 return CargoDirectories {
-                    workspace_root: parent.to_path_buf(),
-                    target_directory: parent_target,
+                    target_directory: target,
+                    workspace_root: manifest_dir,
```

Comment:
I prefer this fast path which traverses upwards indefinitely *if* we validate that there's also a neighbouring `Cargo.toml`.
A pattern I've used is to put all crates related to one project (e.g. all bevy related crates) into one directory, with its own target directory. So the tree looks like the code block. This would break that style of integration (if any bevy crate used `cargo-run-wasm`)
```
.
└── bevy/
    ├── .cargo/config.toml
    ├── target/
    ├── my_games/
    │   ├── dinosaurs/
    │   └── ..
    ├── engine/
    │   ├── bevy/
    │   └── bevy_website/
    └── third_party/
        ├── bevy_mod_on_a_spaceship
        └── bevy_tilemap
```

Reply:
And to clarify, the reason this breaks is that it assumes the workspace root is the top level `bevy/`, instead of say `bevy/engine/bevy/`, which it should be

Reply:
Sorry for the delay, I got caught up in a bunch of life things.

Your example makes sense and is essentially the `// Its possible for this to return false positives if the user leaves a directory named target in the wrong spot.` case that I had neglected to solve.
If I check for both `target` and `Cargo.toml` then your example will fail the fast path and fall back to the slow path, which I am ok with because its definitely a more obscure case.